libs/stats/odc/stats/_cli_save_tasks.py

```diff
@@ -143,11 +154,62 @@ def gqa_predicate(ds):
             predicate=predicate,
             debug=debug,
             msg=on_message,
+            dss=dss, 
+            n_dss=n_dss,
         )
     except ValueError as e:
         print(str(e))
         sys.exit(2)
 
+    if len(products) != 1:
+        for product, count in error_logger.missing_counts.items():
+            print(f"Product {product} has {count} missing datasets.")
+
     if not ok:
         # exit with error code, failure message was already printed
         sys.exit(3)
+
+
+def _parse_products(dc, products, temporal_range):
+        
+    query = {"product": products}
+    query.update(temporal_range.dc_query(pad=0.6)) 
+    dss = ordered_dss(dc, key=lambda ds: (ds.center_time, ds.metadata.region_code), **query)
+    paired_dss = groupby(dss, key=lambda ds: (ds.center_time, ds.metadata.region_code))
+    
+    error_logger = ErrorLogger(products)
+    paired_dss = error_logger.filter(paired_dss)
+    n_dss = max(dataset_count(dc.index, time=query["time"], product=product) for product in products)
```

Comment:
shouldn't that be `min(..)`?

Reply:
I originally had max to give us more info about the number of unpaired datasets but this is unnecessary now - so yeah `min` will give us a tighter upper bound on the number of paired datasets.